localstack/services/sqs/resource_providers/aws_sqs_queue.py

```diff
@@ -181,18 +183,35 @@ def update(
         """
         sqs = request.aws_client_factory.sqs
         model = request.desired_state
-        if request.desired_state.get("QueueName") != request.previous_state.get("QueueName"):
-            # replacement (TODO: find out if we should handle this in the provider or outside of it)
-            # delete old queue
-            sqs.delete_queue(QueueUrl=request.previous_state["QueueUrl"])
-            # create new queue (TODO: re-use create logic to make this more robust, e.g. for auto-generated queue names)
-            model["QueueUrl"] = sqs.create_queue(QueueName=request.desired_state.get("QueueName"))[
-                "QueueUrl"
-            ]
-        else:
-            # TODO: write test with CDK re-deploy updating the queue with no changes to validate this fix
-            model["QueueUrl"] = sqs.get_queue_url(QueueName=model["QueueName"])["QueueUrl"]
 
+        assert request.previous_state is not None
+
+        should_replace = (
+            request.desired_state.get("QueueName", request.previous_state["QueueName"])
+            != request.previous_state["QueueName"]
+        ) or (
+            request.desired_state.get("FifoQueue", request.previous_state["FifoQueue"])
+            != request.previous_state["FifoQueue"]
+        )
```

Comment:
@dominikschubert there must be a way to automate this sort of detection from the spec... :thinking: